src/commands/ls.ts

```diff
@@ -78,7 +102,11 @@ const ls = async (
     print2(
       `Showing${truncationPart} ${label}${postfixPart}.\nResources labeled with * are already accessible to you:`
     );
-    const sortedItems = orderBy(data.items, "isPreexisting", "desc");
+    const sortedItems = orderBy(
+      slice(data.items, 0, requestedSize),
+      "isPreexisting",
+      "desc"
+    );
```

Comment:
Nit: (for clarity and concision):

```suggestion
    const truncated = slice(data.items, 0, requestedSize);
    const sortedItems = orderBy(truncated, "isPreexisting", "desc");
```